xla/service/gpu/fusions/custom.cc

```diff
@@ -180,14 +185,166 @@ absl::StatusOr<BufferAllocation::Slice> GetOperandSlice(
   return absl::InternalError("WTF");
 }
 
+// Gets the innermost while op surrounding the operation `op` if it exists. This
+// also returns nullopt if there are multiple callers for the same while body.
+std::optional<HloInstruction*> GetWhileOp(const HloInstruction& op) {
+  std::unique_ptr<CallGraph> graph = CallGraph::Build(op.GetModule());
```

Comment:
Can this be moved out, so we build the call graph only once? I fear this may cause a performance problem if we process the whole module each time when GetWhileOp is called.